src/error-utils/index.js

```diff
@@ -0,0 +1,92 @@
+import { isEmpty } from 'lodash/fp';
+import joi from 'joi';
+
+const errorDetailsEntrySchema = joi
+  .object()
+  .keys({
+    field: joi.string().allow('', null),
+    value: joi.string().allow('', null),
+    message: joi.string().required(),
+  })
+  .unknown();
+
+const validationErrorDetailsSchema = joi.object().keys({
+  details: joi
+    .array()
+    .min(1)
+    .items(errorDetailsEntrySchema)
+    .required(),
+});
+
+const errorSchema = joi
+  .object()
+  .keys({
+    name: joi.string().required(),
+    message: joi.string().required(),
+    code: joi.string(),
+    details: joi.array().items(errorDetailsEntrySchema),
+  })
+  .unknown();
+
+/**
+ * Validate data with joi
+ * @param {*} data - the data to validate
+ * @param {*} schema - the joi schema
+ * @return {*} the validated data
+ * @throws {Error}
+ */
+const validateWithJoi = (data, schema) => {
```

Comment:
This can be placed in an the `main` utils file since it might be used elsewhere

Reply:
Alternatively this function can be replaced with [`joi.attempt`](https://github.com/hapijs/joi/blob/master/API.md#attemptvalue-schema-message)

Reply:
Done